packages/wallet/src/services/TransactionsTracker.ts

```diff
@@ -101,30 +101,31 @@ export const createAddressTransactionsProvider = ({
                     `Last stored tx: ${lastStoredTransaction?.id} block:${lastStoredTransaction?.blockHeader.blockNo}`
                   );
 
-                const sinceBlock = lastStoredTransaction?.blockHeader.blockNo;
-                const newTransactions = await chainHistoryProvider.transactionsByAddresses({
+                const lowerBound = lastStoredTransaction?.blockHeader.blockNo;
+                const response = await chainHistoryProvider.transactionsByAddresses({
```

Comment:
This looks like we're only fetching the first page of results, whereas, the wallet should still be fetching all results, just one page at a time.

Reply:
How the wallet suppose to change the pagination args of the chain history provider and iterate over pages?

Reply:
We'll need to cover this new behavior in a the wallet e2e too @IvayloAndonov , demonstrating a wallet with txs > page size fetches the complete history. Should be able to set a low page size using `CHAIN_HISTORY_PROVIDER_PARAMS`

Reply:
I'm not sure I understood how to use `CHAIN_HISTORY_PROVIDER_PARAMS` for that purpose since it's from type [HttpProviderConfig](https://github.com/input-output-hk/cardano-js-sdk/blob/dee30b52af5edc1241142a2c06708266a1ae7fa4/packages/cardano-services-client/src/HttpProvider.ts#L11-L44) in most cases to change the provider's `baseUrl`.

Currently [pagination](https://github.com/input-output-hk/cardano-js-sdk/pull/449/files#diff-b015172e50668407821a584b9a03f1c2c70ccf60bd27d3832335c85360d86d96R108) is hardcoded while fetch from the provider as `limit` is 25 ( the default `PAGINATION_PAGE_SIZE_LIMIT` in `cardano-services`)

Should we stick with that value while fetching full tx history sequentially in the loop? @rhyslbw

Reply:
Hardcoded page size is fine for now.

It could also be an optional prop when creating SingleAddressWallet, but I would prefer it to be contained in provider definition. How about `ChainHistoryProvider { maxPageSize: number }`?

Reply:
We already have that [value](https://github.com/input-output-hk/cardano-js-sdk/blob/d267fa0003fbc9d9277f6dc5c226bad2e40a922f/packages/cardano-services/src/ChainHistory/DbSyncChainHistory/DbSyncChainHistoryProvider.ts#L49) provided to the provider from outside, but the case Rhys mentioned is about asserting when txs are > `pagination.limit` , not related with max page size ?

Reply:
> We already have that [value](https://github.com/input-output-hk/cardano-js-sdk/blob/d267fa0003fbc9d9277f6dc5c226bad2e40a922f/packages/cardano-services/src/ChainHistory/DbSyncChainHistory/DbSyncChainHistoryProvider.ts#L49) provided to the provider from outside

We're currently talking about `wallet` package which is client-side, right? The provided value is in cardano-services, which is why I suggested to expose it from the provider.

> So we should somehow to change that limit value in e2e test or to be sure we already have txs greater than 25 in the address?

A unit test in `wallet` package would be sufficient to test this change.

Reply:
> A unit test in wallet package would be sufficient to test this change.

Agree

> We're currently talking about wallet package which is client-side

Aham, as a new public field in [ChainHistoryProvider](https://github.com/input-output-hk/cardano-js-sdk/blob/d267fa0003fbc9d9277f6dc5c226bad2e40a922f/packages/core/src/Provider/ChainHistoryProvider/types.ts#L12) which would be consumed by the wallet for `pagination.limit` value while fetching by pages from provider @mkazlauskas

Reply:
> A unit test in wallet package would be sufficient to test this change.

Yes, you're right @mkazlauskas. That's the better place for this test case

Reply:
> Aham, as a new public field in [ChainHistoryProvider](https://github.com/input-output-hk/cardano-js-sdk/blob/d267fa0003fbc9d9277f6dc5c226bad2e40a922f/packages/core/src/Provider/ChainHistoryProvider/types.ts#L12) which would be consumed by the wallet for pagination.limit value while fetching by pages from provider @mkazlauskas

Yeah. It should probably be an async method like `{ getMaxPageSize(): Promise<number> }` because it can be different for each instance of cardano-services

Reply:
Anyway, as I mentioned above I'd be fine with hardcoding it for this PR and then coming back to it

Reply:
Yes, please create a new task for this @IvayloAndonov.

Reply:
Fetching all transactions, one page at a time is implemented in dd82227870c5992c1b60c0cc491d901674ce2c4a @rhyslbw